### DISCUSSION TITLE
How to resolve this ECMA warning?

### DISCUSSION THREAD

**Discussion by binarykitchen** [web-infra-dev/rsdoctor#572] [General]
```
⚠ Error: [E1004:Warn:ECMA-VERSION-CHECK] The ECMA version used in "index.js" is ES7+, which exceeds the ES5 standard.
```

Not sure where it's coming from, my project. How do you, in general, tell rsbuild to which ECMA version to compile to?

Does it depend on tsconfig where I have this?
```
  "compilerOptions": {
    "module": "esnext",
    "moduleResolution": "bundler",
...
```

**Comment by yifancong** [comment]
This rule cannot currently determine based on config and browserlist; by default, it reports code that is ECMAScript 5 and above. We will optimize this rule in the future. issue：https://github.com/web-infra-dev/rsdoctor/issues/532

**Reply by Verthon** [reply]
Hi @easy1090 I've seen you have optimized this rule here: https://github.com/web-infra-dev/rsdoctor/issues/532

The `.browserslistrc` seems to be ignored by RSDoctor I still see the error:
```shell
warn    Compile Warning: 
⚠ Error: [E1004:Warn:ECMA-VERSION-CHECK] The ECMA version used in "index.0535d4de.js" is ES6, which exceeds the ES5 standard.
```

Despite having the fairly new `.browserslistrc`:

```
> 5%
not dead
```

https://browsersl.ist/#q=%3E+5%25%0Anot+dead

Despite having `.browserslistrc` should we define it manually? 

```
linter: {
        rules: {
          'ecma-version-check': [
            'Warn',
            {
              ecmaVersion: 2015,
              // targets: ["chrome >= 53"],
            },
          ],
        },
      },
```

Versions:

```json
"@rsbuild/core": "^1.2.16",
"@rsbuild/plugin-react": "^1.1.1",
"@rsdoctor/rspack-plugin": "^0.4.13",
```

Here is the repo: https://github.com/Verthon/rsdoctor-ecma-script


Thanks.

**Reply by yifancong** [reply]
@Verthon You need use @rsdoctor/rspack-plugin@1.0, This feature was released on version 1.x.

**Reply by Verthon** [reply]
Thanks that solves the issue, thanks. Congrats with the new major release, great tool 🎉
